lib/autoPagination.js

```diff
@@ -6,42 +6,58 @@ const utils = require('./utils');
 function makeAutoPaginationMethods(self, requestArgs, spec, firstPagePromise) {
   const promiseCache = {currentPromise: null};
   const reverseIteration = isReverseIteration(requestArgs);
-  let listPromise = firstPagePromise;
+  let pagePromise = firstPagePromise;
   let i = 0;
 
-  function iterate(listResult) {
+  function iterate(pageResult) {
     if (
       !(
-        listResult &&
-        listResult.data &&
-        typeof listResult.data.length === 'number'
+        pageResult &&
+        pageResult.data &&
+        typeof pageResult.data.length === 'number'
       )
     ) {
       throw Error(
         'Unexpected: Stripe API response does not have a well-formed `data` array.'
       );
     }
 
-    if (i < listResult.data.length) {
-      const idx = reverseIteration ? listResult.data.length - 1 - i : i;
-      const value = listResult.data[idx];
+    if (i < pageResult.data.length) {
+      const idx = reverseIteration ? pageResult.data.length - 1 - i : i;
+      const value = pageResult.data[idx];
       i += 1;
+
       return {value, done: false};
-    } else if (listResult.has_more) {
+    } else if (pageResult.has_more) {
+      // Search results paginate based on the `next_page` field, while list
+      // calls paginate based on `ending_before` / `starting_after`.
+
       // Reset counter, request next page, and recurse.
       i = 0;
-      const lastId = getLastId(listResult, reverseIteration);
-      listPromise = makeRequest(self, requestArgs, spec, {
-        [reverseIteration ? 'ending_before' : 'starting_after']: lastId,
-      });
-      return listPromise.then(iterate);
+      if (spec.methodType === 'search') {
```

Comment:
nit: this code would feel slightly better to me if the `spec.methodType` check was done once, top-level in `makeAutoPaginationMethods`, rather than each time within `iterate`.

Either something like
```javascript
let getNextPagePromise;
if (spec.methodType === 'search') {
  getNextPagePromise = ...
} else {
  getNextPagePromise = ...
}
function iterate(pageResult) {
  ...
  return getPagePromise(...).then(iterate)
}
```
or

```javascript
let iterate;
if (spec.methodType === 'search') {
  iterate = function (pageResult) {
    ...
  };
} else {
  iterate = ...
}
```
(the second would likely require factoring out a couple helper methods to keep it DRY)

Reply:
Done - good call!